packages/kbn-dev-proc-runner/src/proc.ts

```diff
@@ -87,19 +87,26 @@ export function startProc(name: string, options: ProcOptions, log: ToolingLog) {
 
   const outcome$: Rx.Observable<number | null> = Rx.race(
     // observe first exit event
-    Rx.fromEvent<[number]>(childProcess, 'exit').pipe(
+    Rx.fromEvent<[number, string]>(childProcess, 'exit').pipe(
       take(1),
-      map(([code]) => {
+      map(([code, signal]) => {
         if (stopCalled) {
           return null;
         }
-        // JVM exits with 143 on SIGTERM and 130 on SIGINT, dont' treat then as errors
+
+        // JVM exits with 143 on SIGTERM and 130 on SIGINT, don't treat them as errors
         if (code > 0 && !(code === 143 || code === 130)) {
           throw createFailError(`[${name}] exited with code ${code}`, {
             exitCode: code,
           });
         }
 
+        // A stop signal of SIGABRT indicates a self-inflicted app crash,
+        // then we can't rely on an exit code, because it's not passed
+        if (signal === 'SIGABRT') {
```

Comment:
I'm not 100% confident we can trust that there will not be other signals that might indicate a crash. We don't have to do that in this PR, but in the future I'd like to simplify this code so that it just listens for the rejected promise instead of trying to be so clever and possibly missing some things

Reply:
Fair point, here's a fix: d057ecf